### PR TITLE
Instance LaunchConfig/Template Bug Fix

### PR DESCRIPTION
**Pull request by gambol99**
On merge of https://github.com/kubernetes/kops/pull/6277 the launchconfiguration or template is evaluated; where as before LC was just taken at face value, now the LC/LT is checked for existence. This causes an issue on rolling nodes, nodes where the LC has dissappeared due to retention or terminating instances

### CODE REVIEW

File: upup/pkg/fi/cloudup/awsup/aws_cloud.go [hunk 1]
```diff
@@ -618,10 +617,13 @@ func awsBuildCloudInstanceGroup(c AWSCloud, ig *kops.InstanceGroup, g *autoscali
 			glog.Warningf("ignoring instance with no instance id: %s in autoscaling group: %s", id, cg.HumanName)
 			continue
 		}
-		currentConfigName, err := findInstanceLaunchConfiguration(i)
-		if err != nil {
-			return nil, err
+		// @step: check if the instance is terminating
```
Comment (gambol99) [inline upup/pkg/fi/cloudup/awsup/aws_cloud.go]: Perhaps I should just drop this check? .. its wasn't here before .. @KashifSaadat .. what you reckon?

File: upup/pkg/fi/cloudup/awsup/aws_cloud.go [hunk 2]
```diff
@@ -618,10 +617,13 @@ func awsBuildCloudInstanceGroup(c AWSCloud, ig *kops.InstanceGroup, g *autoscali
...
+		if aws.StringValue(i.LifecycleState) == autoscaling.LifecycleStateTerminating {
+			glog.Warningf("ignoring instance  as it is terminating: %s in autoscaling group: %s", id, cg.HumanName)
+			continue
 		}
+		currentConfigName := findInstanceLaunchConfiguration(i)
```
Comment (gambol99) [inline upup/pkg/fi/cloudup/awsup/aws_cloud.go]: We now take the configuration name regardless, if empty, we just continue and mark as NeedUpdate ... in effect what would happen [here](https://github.com/kubernetes/kops/blob/1.11.1/upup/pkg/fi/cloudup/awsup/aws_cloud.go#L551-L575)

**Comment by gambol99** [comment]
related to https://github.com/kubernetes/kops/issues/6558

File: upup/pkg/fi/cloudup/awsup/aws_cloud.go (hunk 1 above)
Comment (KashifSaadat) [reply upup/pkg/fi/cloudup/awsup/aws_cloud.go]: No I think it's a sensible check to have, as the terminating instance shouldn't be included in the group generated below. Maybe just validate as part of the rolling-update / validation behaviour that this hasn't regressed.

**Comment by gambol99** [comment]
Not sure if the failure is a flack of the fact just not passing the conformance tests anymore ..

**Comment by gambol99** [comment]
But given the last three PRs have the same issue I guessing new checks have been added to sign-storage which is breaking it now :-(

**Pull request merged**